### PR TITLE
MAINT: Move sparsehash code to library_wrappers/sparsehash

### PR DESCRIPTION
**Pull request by gerrymanoim**
First step in https://github.com/quantopian/libpy/issues/172. 

Also good example for how to create a wrapper for libpy. 

TODO: Once agreed, README, etc.

### CODE REVIEW

File: tests/library_wrappers/test_sparsehash.cc [hunk 1]
```diff
@@ -42,4 +43,17 @@ TEST(dense_hash_set, invalid_empty_key) {
     EXPECT_THROW((M8_key{py::datetime64ns::nat()}), std::invalid_argument);
     EXPECT_THROW((M8_key{py::datetime64ns::nat(), 10}), std::invalid_argument);
 }
+
+TEST(any_vtable, map_key) {
```
Comment (llllllllll) [inline tests/library_wrappers/test_sparsehash.cc:47]: Can you move this test back to `test_any.cc` and just use a `std::unordered_map`?
The point of this test is to ensure that `any_vtable` can be used as a map key, which should be true even if you don't use sparsehash.

File: Makefile [hunk 2]
```diff
@@ -150,7 +150,7 @@ GTEST_SRCS := $(wildcard $(GTEST_DIR)/src/*.cc) \
 	$(wildcard $(GTEST_DIR)/src/*.h) $(GTEST_HEADERS)
 GTEST_FILTER ?= '*'
 
-TEST_SOURCES := $(wildcard tests/*.cc)
+TEST_SOURCES := $(wildcard tests/*.cc) $(wildcard tests/library_wrappers/*.cc)
 TEST_DFILES := $(TEST_SOURCES:.cc=.d)
 TEST_OBJECTS := $(TEST_SOURCES:.cc=.o)
 TEST_HEADERS := $(wildcard tests/*.h) $(GTEST_HEADERS)
```
Comment (llllllllll) [inline Makefile:156]: should we add this dir to the test headers?

Comment (gerrymanoim) [reply Makefile:156]: There weren't any headers in there currently, but we could.

File: tests/library_wrappers/test_sparsehash.cc (hunk 1 above)
Comment (gerrymanoim) [reply tests/library_wrappers/test_sparsehash.cc]: whoops - fixed

File: tests/library_wrappers/test_sparsehash.cc [hunk 3]
```diff
@@ -4,9 +4,48 @@
...
+
+namespace test_sparsehash {
+
+using namespace std::literals;
+using namespace py::cs::literals;
+
+class sparsehash_to_object : public with_python_interpreter {};
+
+TEST_F(sparsehash_to_object, sparse_hash_map) {
+    // NOTE: This test takes a long time to compile (about a .5s per entry in this
```
Comment (llllllllll) [inline tests/library_wrappers/test_sparsehash.cc:20]: this comment isn't needed anymore.

File: tests/test_to_object.cc [hunk 4]
```diff
@@ -21,162 +20,29 @@ using namespace py::cs::literals;
...
-    py::owned_ref<PyObject> constref_result = py::to_object(const_ref);
-    check_python_map(constref_result);
-
-    M copy = m;  // Make a copy before moving b/c the lambda above uses ``m``.
-    py::owned_ref<PyObject> rvalueref_result = py::to_object(std::move(copy));
-    check_python_map(rvalueref_result);
-}
-
 TEST_F(to_object, map_to_object) {
     // NOTE: This test takes a long time to compile (about a .5s per entry in this
```
Comment (llllllllll) [inline tests/test_to_object.cc:24]: this comment also doesn't need to be here anymore.

**Pull request merged**